test/run_envoy_tests.sh

```diff
@@ -9,10 +9,16 @@ BINARY_DIR=$2
 # https://bazel.build/versions/master/docs/test-encyclopedia.html. In particular, TEST_SRCDIR is
 # where we expect to find the generated outputs of various scripts preparing input data (these are
 # not only the actual source files!).
-: ${TEST_TMPDIR:=/tmp/envoy_test_tmp}
-: ${TEST_SRCDIR:=/tmp/envoy_test_runfiles}
+# It is a precondition that both $TEST_TMPDIR and $TEST_SRCDIR are empty.
+if [ -z "$TEST_TMPDIR" ] || [ -z "$TEST_SRCDIR" ]
+then
+  TEST_BASE=$(mktemp -d /tmp/envoy_test.XXXXXXXX)
```

Comment:
I switched to `rm -rf`, but only in the default case. I didn't like having a fixed default test directory in `/tmp`, since this plays havoc on multi-user systems or when running multiple test jobs simultaneously. It's a moot point though - when the Bazel works is done, all tests will run under Bazel and `run_envoy_tests.sh` will be basically `bazel test //test...`, so if `rm -rf` makes it easier for you folks, sure.